src/contexts/arrayUtils.js

```diff
@@ -0,0 +1,11 @@
+export function isPresentInArray(arr, item) {
+	let duplicate = false;
+
+	arr.forEach((itr) => {
+		if (itr.productId === item.productId) {
+			duplicate = true;
```

Comment:
I have a suggestion here, if you add one more return duplicate statement below line no 6. , it will save some iterations as it will return as soon as it matches the id

Reply:
We can't break a forEach by returning from the callback function

From MDN docs:

> There is no way to stop or break a forEach() loop other than by throwing an exception. If you need such behavior, the forEach() method is the wrong tool.